spring-cloud-gcp-pubsub/src/main/java/com/google/cloud/spring/pubsub/PubSubAdmin.java

```diff
@@ -235,22 +234,33 @@ public Subscription createSubscription(String subscriptionName, String topicName
 		Assert.hasText(subscriptionName, "No subscription name was specified.");
 		Assert.hasText(topicName, NO_TOPIC_SPECIFIED);
 
-		int finalAckDeadline = this.defaultAckDeadline;
+		Subscription.Builder builder = Subscription.newBuilder()
+				.setName(PubSubSubscriptionUtils.toProjectSubscriptionName(subscriptionName, this.projectId).toString())
+				.setTopic(PubSubTopicUtils.toTopicName(topicName, this.projectId).toString());
+
+		builder.setAckDeadlineSeconds(this.defaultAckDeadline);
 		if (ackDeadline != null) {
 			validateAckDeadline(ackDeadline);
-			finalAckDeadline = ackDeadline;
+			builder.setAckDeadlineSeconds(ackDeadline);
 		}
 
 		PushConfig.Builder pushConfigBuilder = PushConfig.newBuilder();
 		if (pushEndpoint != null) {
 			pushConfigBuilder.setPushEndpoint(pushEndpoint);
+			builder.setPushConfig(pushConfigBuilder);
 		}
+		return createSubscription(builder);
+	}
 
-		return this.subscriptionAdminClient.createSubscription(
-				PubSubSubscriptionUtils.toProjectSubscriptionName(subscriptionName, this.projectId),
-				PubSubTopicUtils.toTopicName(topicName, this.projectId),
-				pushConfigBuilder.build(),
-				finalAckDeadline);
+	/**
+	 * Create a new subscription on Google Cloud Pub/Sub.
+	 *
+	 * @param builder a Subscription.Builder straight from the client API library that exposes all available knobs and
+	 * levers
+	 * @return the created subscription
+	 */
+	public Subscription createSubscription(Subscription.Builder builder) {
+		return this.subscriptionAdminClient.createSubscription(builder.build());
```

Comment:
Should we set the project on the subscription, if it wasn't set?

Reply:
I would say no - I think the point of this method should be straight passthrough. It's main value add is hooking into the credential, not to change the user's inputs.

Reply:
After now working with this, yes, we probably should look for the `projects/` prefix and put one on if missing

Reply:
They require a fully qualified subscription name, right?
If we want to allow "relative" subscription names and topic names, we would need to look for the absense of the `projects/` prefix and add in the configured project to it.